### ISSUE TITLE
rebuildOnUpdate and shouldSwiperUpdate not work properly in version 2.1.1

### ISSUE THREAD

**Issue by damiancompetec** [kidjp85/react-id-swiper#276]
unfortunately those props do not work properly in the latest version.
e.g. I have gallery:
![image](https://user-images.githubusercontent.com/32449459/55358137-f4e2a100-54ce-11e9-9252-3d41823b9b3a.png)
I remove 13'th slide and I get: 
![image](https://user-images.githubusercontent.com/32449459/55358288-4e4ad000-54cf-11e9-922a-ff8cda2a08cc.png)
and now i can't get realIndex = 13 (it is necessary for me, because I use this value to mark current, appropriate thumbnail)

**Comment by ntucakovic** [comment]
How did you remove the 13th slide? I've had an issue with updating slides on a single swiper instance (updating images which are Swiper children, from which I generate slides).

Ended up implementing `useEffect()` which watches for changes on the images array, and there I do `swiperInstance.update()`. Couldn't make use of shouldSwiperUpdate nor rebuildOnUpdate (which shouldn't be used in this case, as the original API supports updating Swiper w/o rebuilding).

**Comment by saintego** [comment]
There seems there is bug in
```
  // Execute each time when props are updated
  useEffect(() => {
    if (swiper !== null) {
      if (rebuildOnUpdate) {
        rebuildSwiper();
      } else if (shouldSwiperUpdate) {
```
`swiper` is always null here

**Comment by pcegarra** [comment]
Same problem here, any idea of how to fix it?

**Comment by dahvinchee** [comment]
adding `observer: true` in your params should do the trick: https://stackoverflow.com/questions/50805404/react-swiper-with-dynamic-content per dimidrol dimidrola answer

**Comment by fkloes** [comment]
I think the reason that `swiper` equals `null` here is, that the swiper instance is not persisted within the component state.

At the top of the function, `swiper` is initialized with `null`. Due to how the effects are defined, it will only be assigned with a swiper reference when the component "mounts". When updated `props` come in, the function runs again, but `swiper` is not assigned with a reference this time when the code from the 2nd effect run.

So would propose to replace
```
let swiper: SwiperInstance = null;
```
with (sorry... i don't know how to annotate this in TS)
```
const [swiper, setSwiper] = useState(null);
```
`buildSwiper()` and `destroySwiper()` now need to invoke `setState()` to maintain the `swiper` reference inside of the component state.

**Comment by kidjp85** [comment]
@All: Sorry for my late response, so busy recently. I made a fix for this issue in v2.3.1.

**Comment by OrenSchwartz** [comment]
im using version 2.3.2 and still encountering this (very annoying :( ) issue.
